lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -255,9 +269,16 @@ def _batched_rows(self, context):
         """
         A generator of batches of (success_rows, error_rows) for this report.
         """
-        for users in self._batch_users(context):
-            users = [u for u in users if u is not None]
-            yield self._rows_for_users(context, users)
+        empty_generator_sentinel = object()
+        batch_users = self._batch_users(context)
+        a_batch_of_users = next(batch_users, empty_generator_sentinel)
+        the_batch_is_empty = a_batch_of_users == empty_generator_sentinel
+        if the_batch_is_empty:
+            yield [], []
+        else:
+            yield self._rows_for_users(context, a_batch_of_users)
+            for users in batch_users:
+                yield self._rows_for_users(context, users)
```

Comment:
Can you elaborate on these 2 statements? Add a comment so that it is helpful for any reader.

Reply:
Sure. I can update the PR with some note.